smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -107,19 +108,23 @@ def deserialize_message(
         input_bytes = [data.blob for data in request.input.data]
         input_meta = [data.tensorDescriptor for data in request.input.data]
 
+    if request.output:
+        output_keys = [tensor_key.key for tensor_key in request.output]
+
     inference_request = InferenceRequest(
         model_key=model_key,
         callback=comm_channel,
         raw_inputs=input_bytes,
-        input_meta=input_meta,
         input_keys=input_keys,
+        input_meta=input_meta,
+        output_keys=output_keys,
         raw_model=model_bytes,
         batch_size=0,
     )
     return inference_request
 
 
-def build_failure_reply(status: "StatusEnum", message: str) -> Response:
+def build_failure_reply(status: "Status", message: str) -> Response:
     return MessageHandler.build_response(
         status=status,  # todo: need to indicate correct status
```

Comment:
Are these TODO comments still applicable?  Can they be addressed or are they follow-up tickets?

Reply:
These are old comments that I just didn't delete. I think these particular ones can be deleted but I wasn't sure if I should be the one to delete them or if others wanted them for reference.

Reply:
Comments removed.